components/link/link.js

```diff
@@ -139,13 +163,24 @@ class Link extends FocusMixin(LitElement) {
 			'truncate': this.lines > 0
 		};
 		const styles = (this.lines > 0) ? { '-webkit-line-clamp': this.lines } : {};
+		const target = this.newWindow && this.target === undefined
+			? '_blank'
+			: this.target;
+		const newWindowIndicator = this.newWindow
+			? html`&nbsp;<d2l-icon class="d2l-new-window" icon="tier1:new-window"></d2l-icon>`
+			: nothing;
+		const newWindowMessage = (target === '_blank')
+			? html`<span class="d2l-offscreen">${this.localize('components.link.open-in-new-window')}</span>`
+			: nothing;
+
 		return html`<a
 				aria-label="${ifDefined(this.ariaLabel)}"
 				class="${classMap(linkClasses)}"
 				style="${styleMap(styles)}"
 				?download="${this.download}"
 				href="${ifDefined(this.href)}"
-				target="${ifDefined(this.target)}"><slot></slot></a>`;
+				target="${ifDefined(target)}"
+			><span style="white-space: nowrap;"><span style="white-space: normal;"><slot></slot></span>${newWindowIndicator}${newWindowMessage}</span></a>`;
```

Comment:
I had to tighten up the whitespace here because it was introducing a lot of spaces every time there was a newline -- one of the visual diff tests caught it.